File: Thirty-Day LeetCoding Challenge/Week 5/Check If a String Is a Valid Sequence from Root to Leaves Path in a Binary Tree/sol.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isValidSequence(TreeNode* root, vector<int>& v) {
        return isValid(root, v, 0);
    }
    bool isValid(TreeNode* root, vector<int>& v, int pos){
        if(root == nullptr || pos >= v.size())
            return false;
        if(root->val != v[pos])
            return false;
        if(root->val == v[pos] && pos == v.size() - 1 && !(root->left) && !(root->right))
            return true;
        return isValid(root->left, v, pos + 1) || isValid(root->right, v, pos + 1);
    }
};

File: 0102. Binary Tree Level Order Traversal/sol.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root is None:
            return []
        cur = [root]
        next = []
        data = []
        ret = []
        
        while len(cur) > 0:
            for node in cur:
                data.append(node.val)
                if node.left is not None:
                    next.append(node.left)
                if node.right is not None:
                    next.append(node.right)
            ret.append(data.copy())
            cur = next
            next = []
            data.clear()
        return ret
            
            